cohere/client.py

```diff
@@ -368,20 +349,43 @@ def summarize(
         return SummarizeResponse(id=response["id"], summary=response["summary"])
 
     def batch_tokenize(self, texts: List[str]) -> List[Tokens]:
-        return [self.tokenize(t) for t in texts]
+        """A batched version of tokenize"""
+        with futures.ThreadPoolExecutor(max_workers=self.num_workers) as executor:
+            res = executor.map(self.tokenize, texts)
+        return list(res)
 
     def tokenize(self, text: str) -> Tokens:
+        """Return a Tokens object of the provided text, see https://docs.cohere.ai/reference/tokenize for advanced usage.
+
+        Args:
+            text (str): Text to summarize.
+        """
         json_body = {'text': text}
-        return Tokens(_future=self._executor.submit(self.__request, cohere.TOKENIZE_URL, json=json_body))
+        res = self.__request(cohere.TOKENIZE_URL, json=json_body)
+        return Tokens(tokens=res['tokens'], token_strings=res['token_strings'])
 
     def batch_detokenize(self, list_of_tokens: List[List[int]]) -> List[Detokenization]:
-        return [self.detokenize(t) for t in list_of_tokens]
+        """A batched version of detokenize"""
+        with futures.ThreadPoolExecutor(max_workers=self.num_workers) as executor:
+            res = executor.map(self.detokenize, list_of_tokens)
+        return list(res)
 
     def detokenize(self, tokens: List[int]) -> Detokenization:
+        """Return a Detokenization object of the provided tokens, see https://docs.cohere.ai/reference/detokenize-1 for advanced usage.
```

Comment:
```suggestion
        """Returns a Detokenization object of the provided tokens, see https://docs.cohere.ai/reference/detokenize for advanced usage.
```
The -1 is a weird artfact